### PR TITLE
43 adapting current fairness metrics to llms

### PR DESCRIPTION
**Pull request by hhaamm**
This adds fairness metrics to LLMs, using the PISA dataset: https://github.com/yunqi-li/Fairness-Of-ChatGPT/tree/main/Data/pisa

The CSV were slightly processed so they have 'H' and 'L' (High and Low) instead of a numeric value in the field readingScore (high is when readingScore >= 500).

For every prompt, the call chooses 4 examples (one for every combination of sensible attribute and label) from the training set and it asks the LLM to fill the answer for one row in the test set.

How to test:

It might require doing `dfx deploy FAI3_backend --mode=reinstall`.

    dfx canister call FAI3_backend calculate_llm_metrics '(1:nat, "pisa": text, 5:nat64, 1:nat32)'

Where the first value is the llm id (the model should exist), '5' if the number of queries (it's recommended that is >= 20 because there are a lot of invalid responses) and '1' is the seed value.

Response example:

```
(
  variant {
    Ok = record {
      metrics = record {
        equal_opportunity_difference = opt vec {
          record { variable_name = "male"; value = -1.0 : float32 };
        };
        statistical_parity_difference = opt vec {
          record { variable_name = "male"; value = -1.0 : float32 };
        };
        precision = opt (1.0 : float32);
        timestamp = 1_741_814_517_794_349_000 : nat64;
        disparate_impact = opt vec {
          record { variable_name = "male"; value = 0.0 : float32 };
        };
        average_odds_difference = opt vec {
          record { variable_name = "male"; value = 0.25 : float32 };
        };
        average_metrics = record {
          equal_opportunity_difference = opt (-1.0 : float32);
          statistical_parity_difference = opt (-1.0 : float32);
          disparate_impact = opt (0.0 : float32);
          average_odds_difference = opt (0.25 : float32);
        };
        recall = opt (0.5 : float32);
        accuracy = opt (0.5 : float32);
      };
      call_errors = 0 : nat32;
      queries = 5 : nat64;
      invalid_responses = 3 : nat32;
    }
  },
)
```

### CODE REVIEW

**Review by LeonardoRyuta — APPROVED** [review]
Canister runs on my machine, and all seems to be working
